crates/napi/src/next_api/project.rs

```diff
@@ -1221,6 +1274,51 @@ pub fn project_update_info_subscribe(
     Ok(())
 }
 
+/// Subscribes to all compilation events that are not cached like timing and progress information.
+#[napi]
+pub fn project_compilation_events_subscribe(
+    #[napi(ts_arg_type = "{ __napiType: \"Project\" }")] project: External<ProjectInstance>,
+    func: JsFunction,
+    event_types: Option<Vec<String>>,
+) -> napi::Result<()> {
+    let turbo_tasks = project.turbo_tasks.clone();
+    let tsfn: ThreadsafeFunction<Arc<dyn CompilationEvent>> =
+        func.create_threadsafe_function(0, |ctx| {
+            let event: Arc<dyn CompilationEvent> = ctx.value;
+
+            let env = ctx.env;
+            let mut obj = env.create_object()?;
+            obj.set_named_property("typeName", event.type_name())?;
+            obj.set_named_property("severity", event.severity().to_string())?;
+            obj.set_named_property("message", event.message())?;
+
+            let external = env.create_external(event, None);
+            obj.set_named_property("eventData", external)?;
+
+            Ok(vec![obj])
+        })?;
+
+    tokio::spawn(async move {
+        let mut receiver = turbo_tasks.get_compilation_events_stream(event_types);
+        loop {
+            match receiver.recv().await {
+                Some(msg) => {
+                    let status = tsfn.call(
+                        Ok(msg),
+                        napi::threadsafe_function::ThreadsafeFunctionCallMode::Blocking,
```

Comment:
Small Nit: Probably just import the `ThreadsafeFunctionCallMode` enum at the top of the file

```suggestion
                        ThreadsafeFunctionCallMode::Blocking,
```